node_modules/oae-transfer/lib/internal/dao.js

```diff
@@ -0,0 +1,778 @@
+                            resolve(managers);
+                        } 
+                    }
+                });
+            });   
+        }); 
+    });
+
+    // Return managers
+    promise.then(function(managers) {
+        return callback(null, managers);
+    }).catch(function() {
+        return callback(null);
+    });
+};
+
+/**
+ * Get the name of the resource by it's id 
+ * take advantage of the differentiation of data type to modify the library
+ * 
+ * @param {Object} ctx
+ * @param {String} idResource
+ * @param {String} principalId
+ * @param {String} principalIdOrigin
+ *
+ * @returns nothing
+ * @api private
+ */
+var _insertIntoLibrary = function (ctx, resource, principalId, principalIdOrigin, callback) {
+    var resourceType = resource.id.split(":")[0];
+    
+    var constant = "";
+    var constantDiscussion = DiscussionsConstants.library.DISCUSSIONS_LIBRARY_INDEX_NAME;
+    var constantFolder = FoldersConstants.library.FOLDERS_LIBRARY_INDEX_NAME;
+    var constantContent = ContentConstants.library.CONTENT_LIBRARY_INDEX_NAME;
+    var constantMeeting = MeetingsConstants.library.MEETINGS_LIBRARY_INDEX_NAME;
+
+    // If the resource is a Content
+    if(resourceType === "c") {     
+        constant = constantContent;
+    // If the resource is a Folder
+    } else if (resourceType === "f") {
+        constant = constantFolder;
+    // If the resource is a Discussion
+    } else if (resourceType === "d") {
+        constant = constantDiscussion;
+    // If the resource is a Discussion
+    } else if (resourceType === "m") {
+        constant = constantMeeting;
+    }
+
+    // Insert to library 
+    _insertOnLibrary(resource, principalId, constant, function(err) {
+        if(err) {
+            return callback(err);
+        }
+
+        // Remove to library 
+        _removeFromLibrary(resource, principalIdOrigin, constant, function(err) {
+            if(err) {
+                return callback(err);
+            }
+            return callback(null);                
+        }); 
+    }); 
+};
+
+/**
+ * Get the name of the resource by it's id 
+ * take advantage of the differentiation of data type to modify the library
+ * 
+ * @param {Object} ctx
+ * @param {String} idResource
+ *
+ * @returns resource
+ * @api private
+ */
+var _getDocument = function (ctx, idResource, callback) {
+    var resourceType = idResource.split(":")[0];
+    // If the resource is a Content
+    if(resourceType === "c") {     
+        ContentDAO.getContent(idResource, function(err, result) {
+            if(err) {
+                return callback(err);
+            }
+            return callback(null, result);        
+        });
+
+    // If the resource is a Folder
+    } else if (resourceType === "g") {
+        _getFoldersByGroupIds(ctx, idResource, function(err, result) {
+            // if err == 1 the resource is a group
+            if(err) {
+                if(err === 1) {
```

Comment:
This is bad practice - the method should not return an error if it's actually the desired result.